src/service/standard-actions-impl.js

```diff
@@ -68,9 +68,38 @@ export class StandardActions {
     /** @const @private {!./viewport/viewport-impl.Viewport} */
     this.viewport_ = Services.viewportForDoc(ampdoc);
 
+    /** @const @private {?Array<string>} */
+    this.ampActionWhitelist_ = null;
+
     this.installActions_(this.actions_);
   }
 
+  /**
+   * Searches for a meta tag containing whitelist of actions on
+   * the special AMP target, e.g.,
+   * <meta name="amp-action-whitelist" content="AMP.setState,AMP.pushState">
+   * @return {?Array<string>} the whitelist of actions on the
+   * special AMP target.
+   * @private
+   */
+  createWhitelist_() {
```

Comment:
How about calling this method `getAmpActionWhitelist`, and return `this.ampActionWhitelist_` if it is already cached (and update the JSDoc accordingly). That way, callers don't need to worry about `this.ampActionWhitelist_` and can just call `this.getAmpActionWhitelist_()` everywhere.